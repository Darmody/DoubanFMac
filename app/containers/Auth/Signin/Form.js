import React, { Component, PropTypes } from 'react';
import { reduxForm } from 'redux-form';

@reduxForm({
  form: 'signin',
  fields: ['email', 'password']
}, state => ({
  logged: state.auth.logged
}))
export default class Form extends Component {
  static propTypes = {
    error: PropTypes.string,
    fields: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    hideModal: PropTypes.func.isRequired,
    logged: PropTypes.bool,
  }

  componentWillReceiveProps(nextProps) {
    if (!this.props.logged && nextProps.logged) {
      this.props.hideModal('signin');
    }
  }

  renderErrorMessage = () => {
    if (this.props.error) {
      return (
        <span className="errorMessage">
          {this.props.error}
        </span>
      );
    }

    return null;
  }

  render() {
    const {
      fields: { email, password, },
      handleSubmit,
    } = this.props;

    return (
      <form onSubmit={handleSubmit}>
        <h2 className="header"> 登录 </h2>
        <div className="field">
          <input type="text" placeholder="邮箱" {...email} />
        </div>
        <div className="field">
          <input type="password" placeholder="密码" {...password} />
        </div>
        <div className="toolbar">
          { this.renderErrorMessage() }
          <button type="submit" className="button">登&nbsp;录</button>
        </div>
      </form>
    );
  }
}
