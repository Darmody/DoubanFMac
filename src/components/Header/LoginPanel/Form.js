// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { Field } from 'redux-form'

const LoginForm = styled.form`
  width: 22.375rem;
  padding: 3.75rem 3.75rem 3.125rem 3.75rem;
`
const Title = styled.h2`
  text-align: center;
  font-size: 1.563rem;
  font-weight: normal;
  margin-bottom: 1.875rem;
  line-height: 1rem;
`

const FieldWrapper = styled.div`
  height: 2.5rem;
  margin-top: .875rem;
  overflow: hidden;
`

type InputProps = {
  className: string,
  input: Object,
  placeholder: string,
  type: string,
}
const ReduxInput = ({
  input,
  type = 'text',
  placeholder = '',
  className = '',
}: InputProps) => (
  <input type={type} placeholder={placeholder} {...input} className={className} />
)
const Input = styled(ReduxInput)`
  border: 1px solid #ccc;
  width: calc(100% - .75rem);
  height: .875rem;
  line-height: .875rem;
  padding: .75rem 0 .75rem .625rem;
  color: #111;
  font-size: .813rem;
  outline: none;
  cursor: text;
  border-radius: 3px;
  box-shadow: inset 0 0 4px rgba(0,0,0,0.06);
`

const Button = styled.input`
  display: block;
  border: none;
  width: 100%;
  height: 2.5rem;
  border-radius: 3px;
  background-color: #59b36a;
  color: white;
  cursor: pointer;
`

type Props = {
  handleSubmit: Function,
}

export default class LoginFormComponent extends PureComponent {
  props: Props

  render() {
    return (
      <LoginForm>
        <Title>欢迎来到豆瓣, 请登录</Title>
        <FieldWrapper>
          <Field
            name="username"
            type="text"
            placeholder="手机号/邮箱/用户名"
            component={Input}
          />
        </FieldWrapper>
        <FieldWrapper>
          <Field
            name="password"
            type="password"
            placeholder="密码"
            component={Input}
          />
        </FieldWrapper>
        <FieldWrapper>
          <Button type="submit" value="登录豆瓣" onClick={this.props.handleSubmit} />
        </FieldWrapper>
      </LoginForm>
    )
  }
}
