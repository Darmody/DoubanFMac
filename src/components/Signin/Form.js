import React from 'react';

const ErrorMessage = ({ error }) => {
  if (error) {
    return (
      <span className="errorMessage">
        {error}
      </span>
    );
  }

  return null;
};

export default ({
  fields: { alias, formPassword, captchaSolution, captchaId },
  error, handleSubmit, captchaCode, fetchCaptcha,
}) => (
  <form onSubmit={handleSubmit}>
    <h2 className="header"> 登录 </h2>
    <input type="hidden" {...captchaId} />
    <div className="field">
      <input type="text" placeholder="邮箱" {...alias} />
    </div>
    <div className="field">
      <input type="password" placeholder="密码" {...formPassword} />
    </div>
    <div className="captchaField">
      <input type="text" className="field" placeholder="验证码" {...captchaSolution } />
      <button onClick={fetchCaptcha}>
        { captchaCode && <img src={`http://douban.fm/misc/captcha?size=m&id=${captchaCode}`} /> }
      </button>
    </div>
    <div className="toolbar">
      <ErrorMessage error={error}/>
      <button type="submit" className="button">登&nbsp;录</button>
    </div>
  </form>
);
