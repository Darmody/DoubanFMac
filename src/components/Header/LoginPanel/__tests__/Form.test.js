import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Form from '../Form'

describe('Component::Login Form', () => {
  test('渲染正确的 DOM 结构', () => {
    const component = renderer.create(<Form handleSubmit={fn => fn} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('登录按钮正确响应 hamdleSubmit 方法', () => {
    const mockSubmit = jest.fn()
    const component = shallow(<Form handleSubmit={mockSubmit} />)

    component.find('[type="submit"]').simulate('click')

    expect(mockSubmit.mock.calls.length).toBe(1)
  })
})
