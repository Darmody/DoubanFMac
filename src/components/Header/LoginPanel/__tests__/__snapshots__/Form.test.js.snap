// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Login Form 渲染正确的 DOM 结构 1`] = `
<form
  className="dyoihW"
>
  <h2
    className="dIBfTJ"
  >
    欢迎来到豆瓣, 请登录
  </h2>
  <div
    className="gfjhTq"
  >
    <div
      className="redux-form-field-mock"
    >
      username
    </div>
  </div>
  <div
    className="gfjhTq"
  >
    <div
      className="redux-form-field-mock"
    >
      password
    </div>
  </div>
  <div
    className="gfjhTq"
  >
    <input
      className="bzvVCm"
      onClick={[Function]}
      type="submit"
      value="登录豆瓣"
    />
  </div>
</form>
`;
